packages/api-electron/src/preload/window.ts

```diff
@@ -145,7 +146,19 @@ class Window implements SSFWindow {
   }
 
   setIcon(icon) {
-    return this.asPromise<void>(this.innerWindow.setIcon, icon);
+    const req = request.defaults({ encoding: null });
+    return new Promise<void>((resolve, reject) => {
+        req.get(icon, (err, res, body) => {
+          const image = nativeImage.createFromBuffer(Buffer.from(body));
+          if (image.isEmpty()) {
+            reject(new Error('Image could not be created from the URL'));
```

Comment:
Currently only electron throws an error when it cannot load the image. Probably worth adding an issue to look into OpenFin throwing this error too.